test/python/qasm/example_version_fail.qasm

```diff
@@ -0,0 +1,11 @@
+OPENQASM 5.0;
```

Comment:
You already test `OPENQASM 3;` so this isn't testing anything new; a good other test would be that minor versions of `2` are also detected, such as `OPENQASM 2.1;` - this would have caught the error with `has_minor_version` above, because the file would have parsed successfully when it should have thrown an error.

Reply:
Eh you got more work than help with my contribution here, sorry for rookie mistakes! Working on it

Reply:
Don't worry about it - it's not really true anyway. Before you made the PR, I didn't know how hard or easy it would be to fix.  After you'd made it, all I had to do was check that what you'd done was sensible, not actually go through and do it myself, and you did this well.

Finding extra special test cases, catching weirdnesses with `re` and things like that is just more experience, and it's why we have review anyway.

Reply:
Well I appreciate the encouragement and would continue to work on this anyways :D